checker/src/org/checkerframework/checker/lock/LockVisitor.java

```diff
@@ -353,9 +576,319 @@ protected boolean skipContractCheck(Tree tree, FlowExpressions.Receiver expr, Fl
+    // to be violated. For example, a method annotated with @ReleasesNoLocks may not
+    // call a method annotated with @MayReleaseLocks.
+    // Also check that matching @GuardSatisfied(index) on a method's formal receiver/parameters matches
+    // those in corresponding locations on the method call site.
+    @Override
+    public Void visitMethodInvocation(MethodInvocationTree node, Void p) {
+
+        SideEffectAnnotation seaOfInvokedMethod = atypeFactory.methodSideEffectAnnotation(TreeUtils.elementFromUse(node), false);
+
+        MethodTree enclosingMethod = TreeUtils.enclosingMethod(atypeFactory.getPath(node));
+
+        ExecutableElement methodElement = null;
+        if (enclosingMethod != null) {
+            methodElement = TreeUtils.elementFromDeclaration(enclosingMethod);
+        }
+
+        SideEffectAnnotation seaOfContainingMethod = atypeFactory.methodSideEffectAnnotation(methodElement, false);
+
+        if (seaOfInvokedMethod.isWeakerThan(seaOfContainingMethod)) {
+            checker.report(Result.failure(
+                    "method.guarantee.violated",
+                    seaOfContainingMethod.getNameOfSideEffectAnnotation(),
+                    methodElement.toString(),
+                    TreeUtils.elementFromUse(node).toString(),
+                    seaOfInvokedMethod.getNameOfSideEffectAnnotation()), node);
+
+            // TODO: issue method.guarantee.violated if a @LockingFree method is synchronized or
+            // contains synchronized blocks.
+        }
+
+        // Check that matching @GuardSatisfied(index) on a method's formal receiver/parameters matches
+        // those in corresponding locations on the method call site.
+
+        Pair<AnnotatedExecutableType, List<AnnotatedTypeMirror>> mfuPair = atypeFactory.methodFromUse(node);
+        AnnotatedExecutableType invokedMethod = mfuPair.first;
+
+        List<AnnotatedTypeMirror> requiredArgs =
+            AnnotatedTypes.expandVarArgs(atypeFactory, invokedMethod, node.getArguments());
+
+        // Index on @GuardSatisfied at each location. -1 when no @GuardSatisfied annotation was present.
+        // Note that @GuardSatisfied with no index is normally represented as having index -1.
+        // We would like to ignore a @GuardSatisfied with no index for these purposes, so if it is encountered we leave its index as -1.
+        // The first element of the array is reserved for the receiver.
+        int guardSatisfiedIndex[] = new int[requiredArgs.size() + 1]; // + 1 for the receiver parameter type
+
+        // Retrieve receiver types from method definition and method call
+
+        guardSatisfiedIndex[0] = -1;
+
+        AnnotatedTypeMirror methodDefinitionReceiver = null;
+        AnnotatedTypeMirror methodCallReceiver = null;
+
+        ExecutableElement invokedMethodElement = invokedMethod.getElement();
+        if (!ElementUtils.isStatic(invokedMethodElement) && !TreeUtils.isSuperCall(node)) {
```

Comment:
You're right. Also, this is wrong because we do need to retrieve the receiver type for calls such as super.myMethod(). Removed this check from here and from one place in LockATF. Thanks.